src/WriteTXT.cpp

```diff
@@ -58,40 +60,58 @@ int libsiedler2::loader::WriteTXT(const boost::filesystem::path& filepath, const
     {
         // "archiviert"
         uint16_t header = 0xFDE7;
+        if(items.size() > std::numeric_limits<uint16_t>::max())
+            return ErrorCode::UNSUPPORTED_FORMAT;
         auto count = static_cast<uint16_t>(items.size());
         uint16_t unknown = 1;
 
         fs << header << count << unknown;
 
-        std::vector<uint32_t> starts(count);
+        std::vector<uint32_t> starts;
+        std::vector<std::pair<std::string, uint32_t>> storedValuesToOffset;
+        starts.reserve(count);
+        storedValuesToOffset.reserve(count);
 
-        uint32_t size = count * sizeof(uint32_t);
-        for(uint32_t i = 0; i < count; ++i)
+        size_t curOffset = count * sizeof(uint32_t);
+        for(const auto& itemPtr : items)
         {
-            const auto* item = static_cast<const ArchivItem_Text*>(items[i]);
-
-            if(item && !item->getText().empty())
+            if(!itemPtr)
+                starts.push_back(0);
+            else
             {
-                starts[i] = size;
-                // add necessary trailing NULL
-                size += static_cast<uint32_t>(item->getFileText(conversion).size() + 1);
+                const auto* item = static_cast<const ArchivItem_Text*>(itemPtr.get());
+                auto text = item->getFileText(conversion);
+                if(text.empty())
+                {
+                    // Only store NULL terminator (empty string)
+                    starts.push_back(curOffset);
+                    storedValuesToOffset.emplace_back(std::string{}, curOffset);
```

Comment:
doenst this store an empty string multiple times? ("without reuse"?)

Reply:
Yes. The original does that so I kept it

Reply:
ah ok. then its fine I guess ;-)